tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -30,3 +40,196 @@ class resource_type(ArmResourceManager.resource_type):
             manager_name='sqlserver',
             annotate_parent=True
         )
+
+
+class BackupRetentionPolicyFilter(Filter):
+
+    schema = type_schema(
+        'backup-retention-policy',
+        **{
+            'op': {'enum': list(scalar_ops.keys())}
+        }
+    )
+
+    def __init__(self, operations_property, retention_limit, data, manager=None):
+        super(BackupRetentionPolicyFilter, self).__init__(data, manager)
+        self.operations_property = operations_property
+        self.retention_limit = retention_limit
+
+    def get_retention_from_policy(self, retention_policy):
```

Comment:
nit: you can use six package to mark function an abstact (will fail to run custodian if not implemented in child classes)